app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -56,7 +56,6 @@ def create
       else
         respond_to do |wants|
           wants.html do
-            flash[:error] = curation_concern.errors.messages[:collections].to_sentence
             build_form
```

Comment:
Flash messages are not used for new work form.  Uses curation_concern.errors instead.  The addition of _form_collections_error.html.erb partial is what actually displays this message.